nexus/types/src/deployment.rs

```diff
@@ -1593,3 +1615,45 @@ mod table_display {
         format!("{value} {UNCHANGED_PARENS}")
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use proptest::prop_assert;
+    use test_strategy::proptest;
+
+    // This test is to ensure that the underlying filter semantics don't change
+    // and break higher level callers. It's fine to do that in some cases, but
+    // at least you'll have to think about it more.
```

Comment:
I don't object to this test (and love proptests in general), but I'm not entirely sure what it's adding - this looks like it's reimplementing `BlueprintZoneDisposition::matches()` with `prop_assert!` instead of bools, right?

Reply:
Yeah I agree with John that a higher-level test (many of which may require #5493) would be more useful.

Reply:
Reverted.